### PR TITLE
add `PercentageSlider`

### PR DESCRIPTION
**Pull request by herbertgithubaccount**
## summary: 
Percentage sliders will provide a way to store numbers in the range 0.0-1.0, but display them in the MCM as 0%-100%. All MCM related parameters/settings will refer to **displayed** values of the slider.

maybe it's worth more clearly indicating that the range doesn't have to be [min, max] (it's just [min, max]), but i wasn't sure how to do this without obfuscating the documentation, and i did say it in a couple places

## status:
documentation and implementation _should_ be done. i also did a fair amount of testing and didn't notice any problems (aside from a few bugs that have already been fixed)

## notes
* in addition to adding the slider, i made one other (relatively minor) change: after adding `PercentageSlider` to  [`autocomplete/.../getComponent.lua`](https://github.com/MWSE/MWSE/pull/479/files#diff-b464da0ef4bb1ece5f602b60cd791ac59f4fd2628bdce5405d45ebc3f0b2d9e9), i reordered some of the other things in in an attempt to group similar components together. i can revert this change if desired.
* there are quite a few file changes/code additions (way more than i was expecting), but pretty much all of it is just documentation: a result of copy-pasting `decimalSlider` autocomplete files (with a few edits!) and rebuilding the docs.
    * not counting the `autocomplete/`, `meta/`, and `docs/` folders:
      * one file was modified: `core/mcm/components/getComponent.lua` (+1 line)
      * one file was added: `core/mcm/components/settings/PercentageSlider.lua` (+62 lines)

## implementation details
the basic idea is:
1. use `decimalSlider` code to handle `config <-> tes3uiSlider` communication

2. use `Slider` code to handle `MCM <-> tes3uiSlider` communication

implementation of the idea:
1. is accomplished by using `decimalSlider` code in places where the `Slider` talks to the user's config (`update` and `enable`). (this accounts for 2 of the 3 overloaded methods in `PercentageSlider`.
2. is accomplished by using `Slider` code in all other functions/utilities. (This is accomplished by inheritance.)

This leaves one overloaded method unaccounted for: the `updateValueLabel` method. This method is pretty much identical to the method in the regular `Slider` class, with one change:
* if `"%s"` is not present in the slider `label`, then `": %s%%"` is added instead of `": %s"`.

### CODE REVIEW

File: autocomplete/definitions/namedTypes/mwseMCMPercentageSlider/createContentsContainer.lua
Comment (C3pa) [inline autocomplete/definitions/namedTypes/mwseMCMPercentageSlider/createContentsContainer.lua:1]: Some of the definition files are unneeded now. You can check in `namedTypes\mwseMCMDecimalSlider`.

**Review by C3pa — COMMENTED** [review]
I am glad you made this. I wanted to do a refactor of the 2 existing slider components to reduce the amount of duplicated code. Hrnchamd explained a good way to do that. I wanted to do it before adding more MCM slider-based settings. This requires some changes to your implementation of PercentageSlider. I didn't want to pull the rug under you, so if you don't feel like applying this refactor, I can do that.

File: misc/package/Data Files/MWSE/core/mcm/components/settings/PercentageSlider.lua [hunk 1]
```diff
@@ -0,0 +1,62 @@
+--[[
+	PercentageSlider:
+		A slider that displays percentages as whole numbers, but stores the values as decimal numbers.
+		i.e., values get shown in the range [0,100], but are stored in the config as [0,1]
+		A Slider Setting that supports decimal numbers with specified
+		amount of decimal places. The underlying implementation has
+		a slider in range (min * 10 ^ decimalPlaces, max * 10 ^ decimalPlaces),
+		while the label above the slider converts that to the decimal.
```
Comment (C3pa) [inline misc/package/Data Files/MWSE/core/mcm/components/settings/PercentageSlider.lua:8]: You can delete this part 😉

**Comment by C3pa** [comment]
Something like this should be enough now:
```lua
--- @class mwseMCMPercentageSlider : mwseMCMSlider
local PercentageSlider = Parent:new()

function PercentageSlider:getNewValue()
	return (self.elements.slider.widget.current + self.min) / 100
end

function PercentageSlider:updateValueLabel()
	local newValue = "" ---@type string|number

	if self.elements.slider then
		newValue = self.elements.slider.widget.current + self.min
	end
	if string.find(self.label, "%s", 1, true) then
		self.elements.label.text = string.format(self.label, newValue)
	else
		self.elements.label.text = string.format("%s: %s%%", self.label, newValue)
	end
end

function PercentageSlider:getCurrentWidgetValue()
	return (self.variable.value * 100) - self.min
end

return PercentageSlider
```

File: autocomplete/definitions/namedTypes/mwseMCMPercentageSlider/createContentsContainer.lua
Comment (herbertgithubaccount) [reply autocomplete/definitions/namedTypes/mwseMCMPercentageSlider/createContentsContainer.lua:1]: i tried to get rid of all the unneeded definition files but let me know if i missed some

**Comment by herbertgithubaccount** [comment]
i refactored the code and updated the documentation. i think my idea of having `min` and `max` refer to MCM values instead of config values makes the code quite a bit more convoluted. i worry it will also make things more confusing for end-users, but i don't have enough experience to make a proper judgement about that. let me know what you think.

in the meantime, i tried to thoroughly document the code (it helped me keep track of what was going on and i left it in because i thought it might be helpful for the PR). feel free to delete it as you wish, there are probably too many comments in the code as it is now.

**Comment by C3pa** [comment]
Added by #482

**Pull request closed without merging**